src/test/ui/parser/increment.stderr

```diff
@@ -0,0 +1,47 @@
+error: Rust has no postfix increment operator
+  --> $DIR/increment.rs:5:6
+   |
+LL |     i++;
+   |      ^^ not a valid postfix operator
+   |
+help: use `+= 1` instead
+   |
+LL |     { let tmp = i; i += 1; tmp };
+   |     +++++++++++  ~~~~~~~~~~~~~~~
```

Comment:
This is because of the code I commented on above where I don't know how to tell if this previous token is a semicolon. I'm not even certain if that's a good heuristic, because if we're at the beginning of a function, the previous token will be a `{`. What I really want is to know if are at the beginning of a new statement. Is there any way to determine that?

Reply:
I don't think there's a great way of knowing that, outside of keeping track of more context - it can be left for a follow-up and we can consider the suggestion possibly correct.

Reply:
Well, it is actually "correct", just unnecessarily complicated.

Reply:
See https://github.com/rust-lang/rust/pull/88672#discussion_r809538489.